### PR TITLE
Docs for template function "index" on lists

### PR DESCRIPTION
**Pull request by colbyguan**
Hello, I noticed the docs for the `index` function were missing. From my usage of it and the helm/sprig source, I assume it comes from [Go templates](https://pkg.go.dev/text/template#hdr-Functions). Because it doesn't come from Sprig, it looks like there is no `mustIndex`. Thought I'd add it in spirit of this comment from https://github.com/helm/helm-www/issues/489#issuecomment-582958396 , with the main tracking issue being https://github.com/helm/helm-www/issues/706 . I am open to whatever wording changes the maintainers feel would be needed!

> We should not expect so many people to know Go templates and sprig functions.

### CODE REVIEW

**Comment by colbyguan** [comment]
hello owners @bridgetkromhout @mattfarina , hoping for a review on this!

File: content/en/docs/chart_template_guide/function_list.md [hunk 1]
```diff
@@ -1562,6 +1562,12 @@ $copy := compact $list
 `compact` panics if there is a problem and `mustCompact` returns an error to the
 template engine if there is a problem.
 
+### index
+
+To get the nth element of a list, use `index list [n]`. To index into multi-dimensional lists, use `index list [n] [m] ...`
```
Comment (bacongobbler) [inline content/en/docs/chart_template_guide/function_list.md:1567]: Line should be hard wrapped at 100 characters as per the rest of the document

**Review by bacongobbler — CHANGES_REQUESTED** [review]
One small change requested.

This is documented as part of Go's text/template docs, though calling it out a second time doesn't hurt I guess.

https://pkg.go.dev/text/template#hdr-Functions

File: content/en/docs/chart_template_guide/function_list.md [hunk 2]
```diff
@@ -1562,6 +1562,13 @@ $copy := compact $list
 `compact` panics if there is a problem and `mustCompact` returns an error to the
 template engine if there is a problem.
 
+### index
+
+To get the nth element of a list, use `index list [n]`. To index into multi-dimensional lists, use
```
Comment (bacongobbler) [inline content/en/docs/chart_template_guide/function_list.md:1567]: my apologies. The document is hard-wrapped at 80 characters. Can you please amend your commit?

**Review by bacongobbler — CHANGES_REQUESTED** [review]

File: content/en/docs/chart_template_guide/function_list.md (hunk 2 above)
Comment (colbyguan) [reply content/en/docs/chart_template_guide/function_list.md]: gotcha, amended now

**Review by bridgetkromhout — APPROVED** [review]

**Comment by bridgetkromhout** [comment]
Thanks!

**Pull request merged**